src/ast/rewriter/bv_rewriter.cpp

```diff
@@ -2803,7 +2803,26 @@ br_status bv_rewriter::mk_ite_core(expr * c, expr * t, expr * e, expr_ref & resu
     return BR_FAILED;
 }
 
-br_status bv_rewriter::mk_bvsmul_no_overflow(unsigned num, expr * const * args, bool is_overflow, expr_ref & result) {
+br_status bv_rewriter::mk_distinct(unsigned num_args, expr * const * args, expr_ref & result) {
+    if (num_args == 0) {
+        return BR_FAILED;
+    }
+    unsigned sz = get_bv_size(args[0]);
+    // check if num_args > 2^sz
+    bool exact = true;
+    do {
+        exact &= (num_args % 2) == 0;
+        num_args /= 2;
+        sz--;
+    } while (num_args > 1 && sz > 0);
```

Comment:
convention: while on the next line after }. 
I know this is contrary to some automatic formatting modes